lib/Benchmark/Executor/template/microtime.template

```diff
@@ -56,30 +56,47 @@ echo serialize([
     'buffer' => $buffer,
 ]);
 
-function benchmark($benchmark, $parameters)
+function benchmark($benchmark, $parameters) : float
 {
+    $time = 0.00;
+
     // run the benchmarks: note that passing arguments to the method slightly increases
     // the calltime, so we explicitly do one thing or the other depending on if parameters
     // are provided.
     if ($parameters) {
-        $startTime = microtime(true);
-
         for ($i = 0; $i < {{ revolutions }}; $i++) {
+            $startTime = microtime(true);
+            
             $benchmark->{{ subject }}($parameters);
-        }
 
-        $endTime = microtime(true);
-    } else {
-        $startTime = microtime(true);
+            $endTime =  microtime(true);
```

Comment:
is there a reason for putting the time measurement in the loop?

I think the idea here is that the SUT might not even take a micro-second, so by putting it outside the loop we are in a better position to judge the average, having it in the loop means we have a larger risk of an inaccurate reading (also `microtime` is not costless).

Reply:
Ah, I see it's fundamental to this approach. I'm not sure this is the best thing to do, would need to do some comparisons to be sure.

Reply:
It might make more sense to have another executor (and another template) which is optimised for memory.

Reply:
Hi,

Yes, I don't see how to simply manage it out of the loop for this approach. However, I don't believe in the risks mentioned, it seems unlikely to me (or it's that the measurements are empty!).

In the current case, the indicated memory already does not correspond to any reality, because in real life the Garbage Collection is activated. The behavior of disabling it is only one way to reduce uncontrollable hazards during the time of a measurement of execution time.
On the other hand, not compensating for this defect by cleaning between measures seems to me to form an aggravation of this defect, whereas there is no longer any pragmatic use in doing so.
Moreover, in the case of several Revolutions, the Memory Peak no longer has any connection with reality, giving delirious scores and therefore has no interest.

In addition, for testing an application like mine: likely to manipulate a lot of data. Not applying this correction would form a default behaviour that would tend to greatly exaggerate its memory consumption, and inexorably tend towards an execution error due to lack of memory.

To summarize: the current behaviour is misleading (on the memory used) and tends to run errors in some cases. The proposed correction does not disturb the measurement of time (except in cases where I would not have understood?) which remains constant, while proposing a more realistic default behaviour in terms of memory and limiting the risk of execution error due to lack of memory.

Another point on the other hand: I put the function cleanGarbageCollectionAndDisable here because of lack of idea. But perhaps it would be more relevant elsewhere.

Reply:
> The proposed correction does not disturb the measurement of time

I think the effect on time would have on "micro" benchmarks, i.e. benchmarks that are best measured in microseconds rather than milliseconds (e.g. benchmarking a function call, or an algorithm). I would have to test to be sure, but I suspsect this will have an effect.

If it does, it might make more sense to have "memory-centric" and "time-centric" executors.